lib/go/csi/csi.pb.go

```diff
@@ -1800,33 +1800,160 @@ func (m *VolumeMetadata) GetValues() map[string]string {
 type DeleteVolumeRequest struct {
 	// The API version assumed by the CO. This field is REQUIRED.
 	Version *Version `protobuf:"bytes,1,opt,name=version" json:"version,omitempty"`
-	// The ID of the volume to be deprovisioned. This field is REQUIRED.
-	VolumeId *VolumeID `protobuf:"bytes,2,opt,name=volume_id,json=volumeId" json:"volume_id,omitempty"`
-	// The metadata of the volume to be deprovisioned. This field is
-	// OPTIONAL.
-	VolumeMetadata *VolumeMetadata `protobuf:"bytes,3,opt,name=volume_metadata,json=volumeMetadata" json:"volume_metadata,omitempty"`
+	// Either `name` or `reference` SHALL be specified by the CO. A CO MAY specify
+	// the `name` when the volume ID and metadata are unknown. For example, in the
+	// case where a `CreateVolume` RPC fails and the CO is unsure of the completion
+	// state of the request AND no longer needs the possibly-created volume.
+	// Alternatively, a CO MAY specify the `reference` field (ID and metadata) when
+	// such information is available to it.
+	// This field is REQUIRED.
+	//
+	// Types that are valid to be assigned to Handle:
+	//	*DeleteVolumeRequest_Name
+	//	*DeleteVolumeRequest_Reference_
+	Handle isDeleteVolumeRequest_Handle `protobuf_oneof:"handle"`
 }
 
 func (m *DeleteVolumeRequest) Reset()                    { *m = DeleteVolumeRequest{} }
 func (m *DeleteVolumeRequest) String() string            { return proto.CompactTextString(m) }
 func (*DeleteVolumeRequest) ProtoMessage()               {}
 func (*DeleteVolumeRequest) Descriptor() ([]byte, []int) { return fileDescriptor0, []int{12} }
 
+type isDeleteVolumeRequest_Handle interface {
+	isDeleteVolumeRequest_Handle()
+}
+
+type DeleteVolumeRequest_Name struct {
+	Name string `protobuf:"bytes,2,opt,name=name,oneof"`
+}
+type DeleteVolumeRequest_Reference_ struct {
+	Reference *DeleteVolumeRequest_Reference `protobuf:"bytes,3,opt,name=reference,oneof"`
+}
+
+func (*DeleteVolumeRequest_Name) isDeleteVolumeRequest_Handle()       {}
+func (*DeleteVolumeRequest_Reference_) isDeleteVolumeRequest_Handle() {}
```

Comment:
this is a pretty ugly type name. does anyone care about this? if so, want to suggest a better field name for `reference` than what's been proposed here?